viewing_party/party.py

```diff
@@ -1,23 +1,181 @@
+
+    for genre, num in genre_dict.items():
+        if num > highest_num:
+            highest_num = num
+            highest_genre = genre
+    
+    return highest_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_friend_movies(friend_list):
+    friend_watched_movies = []
+
+    for movies in friend_list:
+        for movie in movies["watched"]:
+            friend_watched_movies.append(movie)
+    
+    return friend_watched_movies
+
+def get_unique_watched(user_data):
+    watched_movies = user_data["watched"].copy()
+    friend_watched_movies = get_friend_movies(user_data["friends"])
+    not_watched_by_friends = []
+
+    fw_copy = friend_watched_movies.copy()
+
+    for movie in watched_movies:
+        if movie not in friend_watched_movies:
+            not_watched_by_friends.append(movie)
+
+    return not_watched_by_friends
+
+def get_friends_unique_watched(user_data):
+    watched_movies = user_data["watched"].copy()
+    friend_watched_movies = get_friend_movies(user_data["friends"])
+    not_watched_by_user = []
+    
+    for movies in user_data["friends"]:
+        for movie in movies["watched"]:
+            friend_watched_movies.append(movie)
+
+    fw_copy = friend_watched_movies.copy()
+
+    for movie in friend_watched_movies:
+        if movie not in watched_movies and movie not in not_watched_by_user:
+            not_watched_by_user.append(movie)   
+
+    return not_watched_by_user
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended_movies = []
+    friends_unique_watched = get_friends_unique_watched(user_data)
+    friend_watch = get_friend_movies(user_data["friends"])
+
+    if len(user_data["watched"]) == 0 or len(friend_watch) == 0:
+        return []
+
+    for movie in friends_unique_watched:
+        if movie["host"] in user_data["subscriptions"]:
+            recommended_movies.append(movie)
+
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_user_genre(movie_list):
+    user_genre = {}
+
+    if not movie_list:
+        return None
+
+    for movie in movie_list:
+        if movie["genre"] not in user_genre:
+            user_genre[movie["genre"]] = 1
+        else:
+            user_genre[movie["genre"]] += 1
+
+    most_common_genre = max(user_genre, key=user_genre.get)
+
+    return most_common_genre
+
+def get_new_rec_by_genre(user_data):
+    recommended_movies_genre = []
+    most_common_genre = get_user_genre(user_data["watched"])
+    available_recs = get_available_recs(user_data)
+    friend_watch = get_friend_movies(user_data["friends"])
+
+    if len(user_data["watched"]) == 0 or len(friend_watch) ==0:

```

Comment:
Let's move this to the top!

Reply:
Oops, careful with your consistency! 
```suggestion
    if len(user_data["watched"]) == 0 or len(friend_watch) == 0:
```